kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/ProduceAction.java

```diff
@@ -84,11 +84,13 @@ public final class ProduceAction {
   private final Provider<SchemaManager> schemaManagerProvider;
   private final Provider<RecordSerializer> recordSerializerProvider;
   private final Provider<ProduceController> produceControllerProvider;
-  private final Provider<ProducerMetrics> producerMetrics;
+  private final Provider<ProducerMetrics> producerMetricsProvider;
   private final StreamingResponseFactory streamingResponseFactory;
   private final ProduceRateLimiters produceRateLimiters;
   private final ExecutorService executorService;
 
+  private ProducerMetrics metrics = null;
```

Comment:
I did this to start with and it barfed. But, that's because I passed down the provider, not the got ProducerMetrics, which is what I do now.

@rigelbm ready for re-review.